Lib/Rabbit4000/libc/string.c

```diff
@@ -184,18 +238,32 @@ __nodebug
 char far *_f_strcpy(char far *dst, const char far *src)
 {
 	#asm
+#if _RAB6K
 	ld		pz, px					; PZ = PX = dst (saved for return value)
 	ld		py, (sp+@sp+src)		; PY = src
+    clr     hl                      ; set hl to 0 as ps+hl is quicker than ps+0
 	.cpy:
-	ld		a, (py+0)				; *dst = *src
-	ld		(px+0), a				;  (strcpy includes nul terminator)
+	ld		a, (py+hl)
+    pldi              				; *dst++ = *src++
+	or		a							; end of src string?
+	jr		nz, .cpy					; if no (Zero flag reset), go copy another char
+
+	ld		px, pz					; PX = dst (restore the return value)
+#else
+	ld		pz, px					; PZ = PX = dst (saved for return value)
+	ld		py, (sp+@sp+src)		; PY = src
+    clr     hl
+	.cpy:
+	ld		a, (py+hl)				; *dst = *src
+	ld		(px+hl), a				;  (strcpy includes nul terminator)
 	ld		py, py+1					; dst++

```

Comment:
A minor nit, but you could potentially move common code outside of the `#if _RAB6K` block, so it's clear that both platforms have the same entry and exit code, and it's just that non-6000 needs to spell out the `pldi` operation.  Doing so would make it easier to visually compare and confirm the code.